### PR TITLE
[#90] File attachment option for item

### PR DESCRIPTION
**Pull request by pritam1994**

### CODE REVIEW

File: test/models/document_test.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (nbanerjee) [inline test/models/document_test.rb]: Remove all unwanted files

File: app/views/items/_form.html.erb [hunk 2]
```diff
@@ -46,4 +46,12 @@
   <div class="col-sm-4"><%= f.label :employee_id , "Employee:" %></div>
   <div class="col-sm-8"><%= f.collection_select :employee_id, Employee.all, :id, :name_or_email, { prompt: 'Select' }, class: 'form-control' %></div>
 </div>
+<div class="form-group row">
+  <%= f.fields_for :documents do |docs| %>
```
Comment (nbanerjee) [inline app/views/items/_form.html.erb]: should be `doc`

File: app/views/items/_form.html.erb [hunk 3]
```diff
@@ -46,4 +46,12 @@
...
+    <% if docs.object.new_record? %>
+      <div class="col-sm-4"><%= docs.label :attachment, "Invoice" %></div>
```
Comment (nbanerjee) [inline app/views/items/_form.html.erb]: Name it `Attachment`

File: db/migrate/20161122073639_create_documents.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class CreateDocuments < ActiveRecord::Migration[5.0]
+  def change
+    create_table :documents do |t|
```
Comment (nbanerjee) [inline db/migrate/20161122073639_create_documents.rb]: Should add a `type` field

File: app/views/items/show.html.erb [hunk 5]
```diff
@@ -31,6 +31,16 @@
       <p>Purchase Note: <span><%= @item.purchase_note %></span></p>
     <% end %>
 
+    <% if @item.documents.present? %>
+      <p>Attachments:
+        <ol>
+          <% @item.documents.each do |document| %>
+            <li><%= link_to document.attachment_file_name, document.attachment.url, :download => document.attachment_file_name %></li>
```
Comment (nbanerjee) [inline app/views/items/show.html.erb]: Remove `=> `

File: app/controllers/items_controller.rb [hunk 6]
```diff
@@ -7,6 +7,7 @@ def index
 
   def new
     @item = Item.new
+    @item.documents.build
```
Comment (nbanerjee) [inline app/controllers/items_controller.rb]: Move it to `form`

File: app/controllers/items_controller.rb [hunk 7]
```diff
@@ -18,6 +19,10 @@ def create
     end
   end
 
+  def edit
+    @item.documents.build
```
Comment (nbanerjee) [inline app/controllers/items_controller.rb]: Move it to `form`

File: db/migrate/20161122073639_create_documents.rb (hunk 4 above)
Comment (pritam1994) [inline db/migrate/20161122073639_create_documents.rb]: attachment_content_type: field is already their

**Pull request merged**